### DISCUSSION TITLE
Single-purpose plugin design

### DISCUSSION THREAD

**Discussion by TomokiMiyauci** [postcss/postcss#1721] [Q&A] [answered]
Plugins that require parsing of properties and values can have a significant impact on performance if the same parsing is done for each plugin.

Heavy processing should be done only once in its entirety, but for a plugin, we want to provide a single function.
For example, I am thinking of a plugin that only compresses `rgba` using [postcss-value-parser](https://github.com/TrySound/postcss-value-parser).

If the parsing is done for each such small plugin, it will affect the performance.

Is there a mechanism to share values between plugins?
Or is there a good way to achieve the above?

**Comment by ai** [comment, marked as answer]
No mechanism for now. We have plans to add value parser into the core, but it will not finished soon.

**Reply by TomokiMiyauci** [reply]
Thanks for the answer.
